blog/2025-07-01-khepri-default/index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Khepri is now the default metadata store"
+tags: ["khepri"]
+authors: [mgary]
+---
+
+Khepri, the new raft-based RabbitMQ metadata store, became fully supported with RabbitMQ 4.0. With the introduction of RabbitMQ 4.2, we consider Khepri to be mature enough to become the default metadata store, especially given its substantial data safety and recovery improvements over Mnesia. We have performed [a number of benchmarks](#khepri-performance-improvements), showing significant performance improvements in many metadata operations.
+
+<!-- truncate -->
+
+## Khepri Feature Flag now stable
+
+The `khepri` feature flag has now been upgraded to `Stable`, meaning it will now be enabled when running the command `rabbitmqctl enable_feature_flag all`, which should be done after every successful minor version upgrade. We expect that all RabbitMQ clusters upgraded to 4.2 will now have the `khepri` feature flag enabled, and it should be enabled prior to upgrading from 4.2 onwards. In other words, starting with RabbitMQ 4.3, the `khepri` feature flag will be `Required`. Should a cluster be upgraded from RabbitMQ 4.2 without the `khepri` feature flag enabled, it will automatically be enabled as part of the upgrade process.
+
+### Feature Flag Subsystem
+
+The RabbitMQ feature flag subsystem was recently improved by introducing a new category of feature flags known as `Soft Required`. If a feature flag is `Soft Required` starting from version `N`, it is automatically enabled once all RabbitMQ nodes are upgraded to version `N` of RabbitMQ. This is a change from the previous behavior of `Required`, where a feature flag that became required in version `N` of RabbitMQ must be enabled before upgrading to version `N`. It remains best practice to enable feature flags as soon as they become `Stable`, generally immediately after a successful upgrade by running the command `rabbitmqctl enable_feature_flag all`. Nonetheless, we view the introduction of `Soft Required` feature flags as an improvement in user experience, as any required feature flags not already enabled will be automatically enabled when required.
```

Comment:
Not exactly true: a soft required feature flag will be enabled with the first node starting with the new version requiring it because it doesn't know how to run without it.